Library/Homebrew/cmd/update.sh

```diff
@@ -302,6 +312,9 @@ EOS
   # kill all of subprocess on interrupt
   trap '{ pkill -P $$; wait; exit 130; }' SIGINT
 
+  HOMEBREW_UPDATE_FAILED_FILE="$HOMEBREW_REPOSITORY/.git/UPDATE_FAILED"
+  rm -rf "$HOMEBREW_UPDATE_FAILED_FILE"
```

Comment:
Any reason this couldn't be `rm -f "$HOMEBREW_UPDATE_FAILED_FILE"` per the discussion we had the other day?